core/src/test/java/org/apache/iceberg/TestSplitPlanning.java

```diff
@@ -218,6 +237,17 @@ public void testSplitPlanningWithOffsets() {
         "We should get one task per row group", 32, Iterables.size(scan.planTasks()));
   }
 
+  @Test
+  public void testAdaptiveSplitPlanningWithOffests() {
+    table.updateProperties().set(TableProperties.ADAPTIVE_SPLIT_PLANNING, "true").commit();
+
```

Comment:
I think we should have some tests here that show that we are doing the correct thing even if not enough splits are available. For example if we have 1 file with a single offset.